x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/lucene/read/ValuesFromSingleReader.java

```diff
@@ -89,19 +81,20 @@ public int get(int i) {
         }
     }
 
-    private void loadFromSingleLeaf(Block[] target, BlockLoader.Docs docs) throws IOException {
+    private void loadFromSingleLeaf(long jumboBytes, Block[] target, ValuesReaderDocs docs, int offset) throws IOException {
         int firstDoc = docs.get(0);
```

Comment:
I think `firstDoc` should be `int firstDoc = docs.get(offset);`. Our tests may not catch this - there is no correctness issue, but performance could be degraded because we have to reload stuffs.